### PR TITLE
DSD-1445: Updating Storybook documentation for Modal component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1445](https://jira.nypl.org/browse/DSD-1445)

## This PR does the following:

- Updates the `Modal` component Storybook documentation to version 7.
- Compare against prod: https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-overlays-switchers-modal--modal-trigger

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Modal/Modal.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+## ModalTrigger
+
+<Description of={ModalTrigger} />
+
+<Source
+  code={`
+import { ModalTrigger } from "@nypl/design-system-react-components";
+
+//...
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.mdx:42]: We've taken this out of other code examples.  For consistency, should we do that here as well?

File: src/components/Modal/Modal.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+number of elements that use the `onClick` attribute. The `onClose` function can
+be used on any element that uses the `onClick` attribute.
+
+**NOTE: in this case, the `buttonText` prop is not necessary since you have to
+create and render your own button.**
+
+<Source
+  code={`
+import { useModal } from "@nypl/design-system-react-components";
+// ...
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.mdx:80]: Same here.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  Just some questions about how the code examples are structured.

File: src/components/Modal/Modal.mdx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Modal/Modal.mdx:42]: I think the `Modal` components are special enough that it does need this example snippet.

**Pull request merged**